lib/wings/hydra/works/services/add_file_to_file_set.rb

```diff
@@ -0,0 +1,54 @@
+# TODO: This should live in Hyrax::AddFileMetadataToFileSet service and should work for all valkyrie adapters.
+module Wings::Works
+  class AddFileToFileSet
+    class << self
+      # Adds a file to the file_set
+      # @param file_set [Valkyrie::Resource] adding file to this file set
+      # @param [IO,File,Rack::Multipart::UploadedFile, #read] file the object that will be the contents.
+      #   If file responds to :mime_type, :content_type, :original_name, or :original_filename, those will be called to provide metadata.
+      # @param [RDF::URI or String] type URI for the RDF.type that identifies the file's role within the file_set
+      # @param [Boolean] update_existing whether to update an existing file if there is one. When set to true, performs a create_or_update.
+      #   When set to false, always creates a new file within file_set.files.
+      # @param [Boolean] versioning whether to create new version entries (only applicable if +type+ corresponds to a versionable file)
+      def call(file_set:, file:, type:, update_existing: true, versioning: true)
+        raise ArgumentError, 'supplied object must be an instance of Valkyrie::Resource' unless file_set.is_a?(Valkyrie::Resource) && file_set.file_set?
+        raise ArgumentError, 'supplied file must respond to read' unless file.respond_to? :read
+
+        af_type = normalize_type(type)
+        raise ArgumentError, "supplied type (#{type}) is not supported" unless af_type.present?
+
+        # TODO: required as a workaround for https://github.com/samvera/active_fedora/pull/858
```

Comment:
can you clarify the nature of the work-around?

is this necessary in all cases, or only for certain object states?

Reply:
That got copied over from Hydra::Works and really doesn't apply here with the simple wrapper.  It applies in the Hydra::Works method being wrapped.  I removed it here.

https://github.com/samvera/hydra-works/blob/master/lib/hydra/works/services/add_file_to_file_set.rb#L14